backup/moodle2/restore_cms_stepslib.php

```diff
@@ -89,13 +89,11 @@ protected function process_cms_types(array $data) {
 
         $typeid = $this->find_existing_cms_type($data);
 
-        if ($typeid === false) {
-            $typeid = $DB->insert_record('cms_types', $data);
+        // If we can find a type ID, we can restore the mapping. Otherwise do nothing.
+        if ($typeid !== false) {
+            $this->set_mapping('cms_types', $oldid, $typeid, true);
+            $DB->update_record('cms', (object) ['id' => $this->get_new_parentid('cms'), 'typeid' => $typeid]);
```

Comment:
Shouldn't we always refer to idnumber regardless of if it's a same site backup or a backup from a different site?

Reply:
This is the standard way it is done. Do you want me to change it?

Reply:
I thought the idea here was that when restoring, we wanted to match on idnumber since typeid might be different (e.g. from a different env, or someone deleted the old cms type and imported/created a new one, etc)

Reply:
On the same site then it wold be using the same ID. If the type gets deleted, then the ID would not be found and it falls back to looking for id number (ugh! that is so confusing!).

Reply:
Yeah, I think for consistency sake (if it be possible) we should just always match on idnumber, so your id mapping thing should look for cms_type with the same idnumber

Reply:
done.